server/src/main/java/com/flipkart/varadhi/RestVerticle.java

```diff
@@ -10,29 +13,44 @@
 import io.vertx.ext.web.Router;
 import lombok.extern.slf4j.Slf4j;
 
+import java.util.List;
+import java.util.Map;
+
 @Slf4j
 public class RestVerticle extends AbstractVerticle {
-
-    private final ServerConfiguration configuration;
-
-    private final CoreServices coreServices;
+    private final List<RouteDefinition> apiRoutes;
+    private final Map<RouteBehaviour, RouteBehaviourProvider> behaviorProviders;
 
     private HttpServer httpServer;
 
-    public RestVerticle(ServerConfiguration configuration, CoreServices coreServices) {
-        this.configuration = configuration;
-        this.coreServices = coreServices;
+    public RestVerticle(
+            List<RouteDefinition> apiRoutes, Map<RouteBehaviour, RouteBehaviourProvider> behaviorProviders
+    ) {
+        this.apiRoutes = apiRoutes;
+        this.behaviorProviders = behaviorProviders;
     }
 
     @Override
     public void start(Promise<Void> startPromise) {
+
         log.info("HttpServer Starting.");
         Router router = Router.router(vertx);
 
         FailureHandler failureHandler = new FailureHandler();
-        for (RouteDefinition def : coreServices.getRouteDefinitions()) {
+        for (RouteDefinition def : apiRoutes) {
             Route route = router.route().method(def.method()).path(def.path());
-            def.behaviours().forEach(d -> d.Configure(route, def, coreServices));
+            def.behaviours().forEach(behaviour -> {
```

Comment:
This may not be safe right? I think the handlers need to be configured in fixed order.

For eg: BodyHandler should not come before Auth handlers.